kubernetes/data_source_kubernetes_service_account_test.go

```diff
@@ -0,0 +1,51 @@
+package kubernetes
+
+import (
+	"fmt"
+	// "regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccKubernetesDataSourceServiceAccount_basic(t *testing.T) {
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesDataSourceServiceAccountConfig_basic(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.annotations.%", "2"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.annotations.TestAnnotationOne", "one"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.annotations.TestAnnotationTwo", "two"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.labels.%", "3"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.labels.TestLabelOne", "one"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.labels.TestLabelTwo", "two"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.labels.TestLabelThree", "three"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "metadata.0.name", name),
+					resource.TestCheckResourceAttrSet("data.kubernetes_service_account.test", "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_service_account.test", "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_service_account.test", "metadata.0.self_link"),
+					resource.TestCheckResourceAttrSet("data.kubernetes_service_account.test", "metadata.0.uid"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "secret.#", "2"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "image_pull_secret.#", "2"),
+					resource.TestCheckResourceAttr("data.kubernetes_service_account.test", "automount_service_account_token", "false"),
```

Comment:
Nice! Looks like this test is passing.

```
[dakini@dax terraform-provider-kubernetes (review_731)]$ cd kubernetes ; TF_ACC=true TF_LOG=trace go test -v -run TestAccKubernetesDataSourceServiceAccount_basic ; cd -
--- PASS: TestAccKubernetesDataSourceServiceAccount_basic (0.92s)
PASS
ok  	github.com/terraform-providers/terraform-provider-kubernetes/kubernetes	0.944s
/home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes
```

This file is also really similar to what we have throughout our repos. But I was hoping we can take it a step further to improve the process. I'll show you what I mean. I recently submitted a similar test in a PR and someone showed me a way to make it even better: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/801/commits/c34cc422ad18813870d706d0f41d4adf64baf4eb

Basically we're taking this concept of "DAMP not DRY", and adding a little bit of duplicate code for the sake of readability and making the test more self-contained. This is a description of the concept, in case you're curious:  https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests/11837973#11837973

Could you update this file to have fewer tests and write out the kubernetes_service_account resource instead of referring to `testAccKubernetesServiceAccountConfig_basic`? That would help us in this goal. Thanks! (And you can refer to my PR in this comment for an example, if needed.)

Reply:
Thanks for the DAMP pointer. Indeed, the test looks a lot better now!